src/main/java/org/skriptlang/skript/bukkit/toolcomponent/ToolRuleWrapper.java

```diff
@@ -0,0 +1,72 @@
+package org.skriptlang.skript.bukkit.toolcomponent;
+
+import org.bukkit.Material;
+import org.bukkit.Tag;
+import org.bukkit.inventory.meta.components.ToolComponent.ToolRule;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Map;
+
+public class ToolRuleWrapper implements ToolRule {
+
+	private Collection<Material> blocks = new ArrayList<>();
+	private boolean correct = false;
+	private Float speed = 5.0F;
```

Comment:
these need to reflect default behavior of nullable, these values also feel magic number like where are they taken from?
```suggestion
	private Boolean correct;
	private Float speed;
```